src/PrimDist.hs

```diff
@@ -190,46 +190,46 @@ instance Show ErasedPrimDist where
 sample ::
      PrimDist a
   -> Sampler a
-sample (HalfCauchyDist σ )  =
-  createSampler (sampleCauchy 0 σ) >>= pure . abs
-sample (CauchyDist μ σ )  =
-  createSampler (sampleCauchy μ σ)
-sample (HalfNormalDist σ )  =
-  createSampler (sampleNormal 0 σ) >>= pure . abs
-sample (NormalDist μ σ )  =
-  createSampler (sampleNormal μ σ)
-sample (UniformDist min max )  =
-  createSampler (sampleUniform min max)
-sample (DiscrUniformDist min max )  =
-  createSampler (sampleDiscreteUniform min max)
-sample (GammaDist k θ )        =
-  createSampler (sampleGamma k θ)
-sample (BetaDist α β  )         =
-  createSampler (sampleBeta α β)
-sample (BinomialDist n p  )     =
-  createSampler (sampleBinomial n p) >>=  pure .  length . filter (== True)
-sample (BernoulliDist p )      =
-  createSampler (sampleBernoulli p)
-sample (CategoricalDist ps )   =
-  createSampler (sampleCategorical (V.fromList $ fmap snd ps)) >>= \i -> pure $ fst $ ps !! i
-sample (DiscreteDist ps )      =
-  createSampler (sampleDiscrete ps)
-sample (PoissonDist λ ) =
-  createSampler (samplePoisson λ)
-sample (DirichletDist xs ) =
-  createSampler (sampleDirichlet xs)
+sample (HalfCauchyDist σ )
+  = createSampler (sampleCauchy 0 σ) <&> abs
```

Comment:
I don't think this improves readability significantly, if anything it makes it a bit less readable (same for below)

Reply:
Simpler & by using `fmap` we don't need to compose the function with `pure`. I have switched to using `<$>` instead of its flipped variant `<&>` as `<$>` is more common & in prelude